### PR TITLE
Implement indent for dumps and make sure that all arguments are passed from dump to dumps

### LINKED ISSUE #137: dumps doesn't support indent
As this library should work as "a drop-in replacement for Python's built-in json module" it should support the indent feature for `dumps()`.

```python
import hyperjson

data = {
    'key1': [1,2,3],
    'key2': {
        'a': '1',
        'b': 2
    }
}

text = hyperjson.dumps(data, sort_keys=True, indent=4)
print(text)
```

Running this code results in the error:
```
Traceback (most recent call last):
  File "hyperjson_test.py", line 11, in <module>
    text = hyperjson.dumps(data, sort_keys=True, indent=4)
TypeError: dumps() got an unexpected keyword argument: indent
```

### PR DESCRIPTION
**Pull request by shenek**
closes #137

### CODE REVIEW

**Comment by mre** [comment]
Great effort; I appreciate it. Just a few remarks before we can merge this.

File: tests/test_dumps.py [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+
+
 simple_types = [1, 1.0, -1, None, "str", True, False]
 
 
 @pytest.mark.parametrize("payload", simple_types)
-def test_simple_types(payload):
-    assert json.dumps(payload) == hyperjson.dumps(payload)
+def test_simple_types(payload, dumps):
+    assert dumps(payload) == dumps(payload)
```
[the remark this answers is not in the document]
Comment (shenek) [reply tests/test_dumps.py]: Definitely should. Thanks for noticing.

File: tests/test_dumps.py [hunk 2]
```diff
@@ -48,12 +64,11 @@ def test_simple_dicts(d, allowed):
 
 
 @pytest.mark.parametrize("d", complex_dicts)
-def test_complex_dicts(d):
-    assert ignore_whitespace(json.dumps(
-        d)) == ignore_whitespace(hyperjson.dumps(d))
+def test_complex_dicts(d, dumps):
+    assert ignore_whitespace(dumps(d)) == ignore_whitespace(dumps(d))
```
[the remark this answers is not in the document]
Comment (shenek) [reply tests/test_dumps.py]: Yes. Thanks for noticing.

**Comment by mre** [comment]
Merged! Thanks a lot for adding those features and for the fast turnarounds, @shenek.

**Comment by shenek** [comment]
You're welcome.

**Pull request merged**